test/extended/prometheus/collection_profiles.go

```diff
@@ -92,28 +93,46 @@ var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfil
 		o.Expect(err).To(o.BeNil())
 	})
 
-	var _ = g.Context("in an environment that migrates from default to minimal collection profile", func() {
-		g.It("should apply default collection profile initially", func() {
+	defaultOnlyMetric := "kube_deployment_status_replicas"
+
+	g.Context("initially, in a homogeneous default environment,", func() {
+		g.BeforeAll(func() {
 			o.Eventually(func() error {
-				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileDefault})
+				enabled, err := r.isProfileEnabled(collectionProfileDefault)
```

Comment:
Introduced a `r.isProfileEnabled` to improve resiliency and be order independent (except the `initial` context, as it's reasonable that it executes first (for certain pre-checks), all others can be shuffled).